src/libs/sfdk/device.cpp

```diff
@@ -603,9 +603,15 @@ void DeviceManager::updateDevicesXml() const
         } else {
             EmulatorDevice *const emulatorDevice = static_cast<EmulatorDevice *>(device.get());
             QString mac = EmulatorPrivate::get(emulatorDevice->emulator())->mac();
-            QTC_CHECK(!mac.isEmpty());
-            if (!mac.isEmpty())
-                xmlData.m_index = mac.at(mac.count() - 1);
+            if (QTC_GUARD(mac.length() >= 2)) {
+                auto hex2dec = [](const QString &hex) {
+                    bool ok;
+                    const QString dec = QString::number(hex.toInt(&ok, 16));
+                    QTC_CHECK(ok);
+                    return dec;
+                };
+                xmlData.m_index = hex2dec(mac.right(2));
+            }
```

Comment:
How is this better than:
```
if (QTC_GUARD(mac.length() >= 2)) {
                bool ok;
                xmlData.m_index = QString::number(mac.right(2).toInt(&ok, 16));
                QTC_CHECK(ok);
            }
```
?

Reply:
I started with that version too :) but it does not expresses the intent clearly (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p3-express-intent) and does not support code reuse (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f1-package-meaningful-operations-as-carefully-named-functions).